### DISCUSSION TITLE
Old checks still appearing in PR checks after being removed

### DISCUSSION THREAD

**Discussion by pararang** [community/community#139960] [Actions] [answered]
### Select Topic Area

Question

### Body

I am experiencing an issue with one of my repositories where checks that were previously removed are still showing up in the Pull Request [(PR) discussions](https://github.com/pararang/gostein/pull/33). 

This is preventing the merge process from completing, as the old checks remain listed as required even though they should no longer be relevant.
![Screenshot 2024-09-27 at 22 05 50](https://github.com/user-attachments/assets/b87771d6-1f7c-45cf-9070-ee8811cc3b20)

Here are the steps I followed:

1. I removed the old checks from the Actions/CICD configuration.
2. I ensured that the old checks are no longer present in the .github/workflows configuration files.

![Screenshot 2024-09-27 at 22 06 04](https://github.com/user-attachments/assets/acc46cfa-3888-4086-9dbb-8970be6c61d0)

Despite these changes, the removed checks still appear in the PR discussions and are marked as failing, blocking the merge.
Is there a way to refresh or update the status of these checks so the PR can proceed without issues? Or is there any other configuration that I need to review to ensure that the old checks are completely removed from the PR checks?

Thank you for your assistance and guidance.

**Comment by lovrozagar** [comment, marked as answer]
Hi @pararang,

It sounds like GitHub is still referencing the old checks you’ve removed. This can happen occasionally, but there are a few things you can try to fix it.

First, head to your repository settings and check the Branch Protection Rules. Sometimes, even after removing a check from your workflow files, it might still be listed as required in the branch protection settings. If you see the old checks there, you’ll need to remove them.

If that doesn’t help, you could try re-running the workflows on the pull request. There’s usually a button in the PR checks section that says "Re-run jobs." That can sometimes force GitHub to refresh the status checks.

Another thing to try is pushing a new commit to the PR or even just closing and reopening the pull request. That can trigger GitHub to clear out the old checks and update the status.

Lastly, if you’re using any external CI/CD services, make sure there aren’t any lingering webhooks that might still be reporting those old checks.

Hope this helps! Let me know how it goes or if you need more guidance.

**Reply by pararang** [reply]
cool, thanks @lovrozagar. its as you said, the check is still listed as required in the branch protection settings.

**Comment by 65156** [comment]
is there any way to forcibly remove specific status checks or just reset the cache, we had a lambda function incorrectly publishing a timestamp into the status check so one of our repos has 100s of status checks that are in pending state and never complete... 

just looks botched.

**Reply by robnagler** [reply]
Disable branch protection cleared it for me.
